s2i/docker_client_wrapper.go

```diff
@@ -0,0 +1,62 @@
+package s2i
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+)
+
+// Wrapper to workaround https://github.com/containers/podman/issues/13770
+type podmanDockerClient struct {
+	pimpl client.CommonAPIClient
+}
```

Comment:
```suggestion
type podmanDockerClient struct {
	client.CommonAPIClient
}
```
This will map all fields from the embedded struct to the parent.  Presuming you match pointer receivers if necessary, this should remove the need to manually map every method on the parent to the child, while still allowing an "override" for `ContainerCommit` which exists on the parent.